src/js/textmode.js

```diff
@@ -272,14 +273,24 @@ textmode.create = function (container, options = {}) {
     const aceSession = aceEditor.getSession()
     aceEditor.$blockScrolling = Infinity
     aceEditor.setTheme(this.theme)
-    aceEditor.setOptions({ readOnly: isReadOnly })
     aceEditor.setShowPrintMargin(false)
     aceEditor.setFontSize('14px')
     aceSession.setMode('ace/mode/json')
     aceSession.setTabSize(this.indentation)
     aceSession.setUseSoftTabs(true)
     aceSession.setUseWrapMode(true)
 
+    const aceEditorOptions = {
+      readOnly: isReadOnly
+    }
+
+    if (this.options.allowSchemaSuggestions && this.options.schema) {
```

Comment:
I think setting the `SchemaTextCompleter` here is redundant: at the bottom of `textmode.create` we call `this.setSchema`, which calls `textmode._onSchemaChange`, which also configures the `SchemaTextCompleter`.

Reply:
That is correct - I have removed this config and now it is done only once on `textmode._onSchemaChange`. tnx

Reply:
👌  thanks